mixer/pkg/api/grpcServer.go

```diff
@@ -307,17 +307,29 @@ func (s *grpcServer) Report(ctx context.Context, req *mixerpb.ReportRequest) (*m
 
 		switch req.RepeatedAttributesSemantics {
 		case mixerpb.DELTA_ENCODING:
-			reportBag.Reset()
```

Comment:
Is there a code path where this can be nil?

Reply:
I'm not sure if it's possible this can be nil here because there are many switch and if branches, but adding nil judgment will make the code safer. @mandarjog

Reply:
In my opinion, we'd better add judgment to some places that may cause panic now or in the future because of null pointer, because the code in other places may change， which will also change the code path.